### PR TITLE
Hide System Objects

### PR DESCRIPTION
**Pull request by Weiko**
Following https://github.com/twentyhq/twenty/issues/2449

We now want to use the new isSystem bool to hide system objects from the navbar + view pages.

This PR introduces filters on the object type with isSystem, isCustom and isActive.

### CODE REVIEW

File: server/src/database/typeorm/metadata/migrations/1699890270187-setupMetadataTables.ts [hunk 1]
```diff
@@ -1,36 +1,71 @@
-import { MigrationInterface, QueryRunner } from "typeorm";
+import { MigrationInterface, QueryRunner } from 'typeorm';
 
 export class SetupMetadataTables1699890270187 implements MigrationInterface {
-    name = 'SetupMetadataTables1699890270187'
```
Comment (Weiko) [inline server/src/database/typeorm/metadata/migrations/1699890270187-setupMetadataTables.ts:4]: Simply indentation

File: server/src/metadata/object-metadata/dtos/object-metadata.dto.ts [hunk 2]
```diff
@@ -47,13 +47,13 @@ export class ObjectMetadataDTO {
   @Field({ nullable: true })
   icon: string;
 
-  @Field()
+  @FilterableField()
```
Comment (Weiko) [inline server/src/metadata/object-metadata/dtos/object-metadata.dto.ts:50]: I realized that the filtering was being done on the frontend, which could create issues with pagination. I'm introducing these two filters, along with the 'isSystem' filter, for future improvements.

**Review by charlesBochet — COMMENTED** [review]
Left comment!

File: front/src/modules/object-metadata/graphql/queries.ts [hunk 3]
```diff
@@ -2,7 +2,7 @@ import { gql } from '@apollo/client';
 
 export const FIND_MANY_METADATA_OBJECTS = gql`
   query ObjectMetadataItems {
-    objects(paging: { first: 1000 }) {
+    objects(paging: { first: 1000 }, filter: { isSystem: { is: false } }) {
```
[the remark this answers is not in the document]
Comment (charlesBochet) [reply front/src/modules/object-metadata/graphql/queries.ts:5]: Yep :p
image

[the remark this answers is not in the document]
Comment (Weiko) [reply front/src/modules/object-metadata/graphql/queries.ts]: My bad, I didn't realise it was used for this part as well!

**Review by charlesBochet — APPROVED** [review]
Great!

**Pull request merged**